packages/runtime/container-runtime/src/dataStore.ts

```diff
@@ -202,3 +206,48 @@ class DataStore implements IDataStore {
 		});
 	}
 }
+
+export class DataStore2 implements IDataStore2 {
+	private _dataStore?: IDataStore;
+	private attachedRuntime: boolean = false;
+	public get context(): IFluidDataStoreContextDetached {
+		return this.fluidDatastoreContext;
+	}
+	constructor(
+		private readonly fluidDatastoreContext: LocalDetachedFluidDataStoreContext,
+		private readonly internalId: string,
+		private readonly runtime: ContainerRuntime,
+		private readonly datastores: DataStores,
+		private readonly logger: ITelemetryLoggerExt,
+	) {}
+
+	private get dataStore(): IDataStore {
+		if (this._dataStore === undefined) {
+			assert(this.attachedRuntime, "DataStore2 must be attached to runtime before use");
```

Comment:
Instead of doing this here, you should do this in `attachRuntime`. That way, you don't have to expose `channelValue` on data store context and you won't need to track `this.attachedRuntime`

Reply:
[Created task](https://dev.azure.com/fluidframework/internal/_workitems/edit/6783)